src/plugins/saved_search/public/mocks.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { searchSourceInstanceMock } from '@kbn/data-plugin/common/search/search_source/mocks';
+
+const savedSearchStartMock = () => ({
+  get: jest.fn().mockImplementation(() => ({
+    id: 'savedSearch',
+    title: 'savedSearchTitle',
+    searchSource: searchSourceInstanceMock,
+  })),
+  getAll: jest.fn(),
+  getNew: jest.fn().mockImplementation(() => ({
+    searchSource: searchSourceInstanceMock,
```

Comment:
to fix a lot of jest tests it needs a "real" search source. Here's a function that would generate one (could be prettier of course)

```
const createEmptySearchSource = jest.fn(() => {
  const deps = {
    getConfig: jest.fn(),
  } as unknown as SearchSourceDependencies;
  const searchSource = new SearchSource({}, deps);
  searchSource.fetch$ = jest.fn().mockReturnValue(of({ rawResponse: { hits: { total: 2 } } }));
  searchSource.createChild = jest.fn((options = {}) => {
    const childSearchSource = new SearchSource({}, deps);
    childSearchSource.setParent(searchSource, options);
    childSearchSource.fetch$ = <T>() =>
      of({ rawResponse: { hits: { hits: [] } } } as unknown as IKibanaSearchResponse<
        SearchResponse<T>
      >);
    return childSearchSource;
  });
  return searchSource;
});
```
Without it, every change to the searchSource object is lost ... and when testing the whole state container this causes the failure